lib/rules/arrow-body-style.js

```diff
@@ -65,38 +68,57 @@ module.exports = {
+                        fix(fixer) {
+                            if (blockBody.length !== 1 || blockBody[0].type !== "ReturnStatement") {
+                                return null;
+                            }
 
-                    if (asNeeded && requireReturnForObjectLiteral && blockBody[0].type === "ReturnStatement" &&
-                        blockBody[0].argument.type === "ObjectExpression") {
-                        return;
-                    }
+                            const returnValue = blockBody[0].argument;
+                            const sourceText = sourceCode.getText();
+                            const textBeforeReturn = sourceText.slice(arrowBody.range[0] + 1, blockBody[0].range[0]);
+                            const textBetweenReturnAndValue = sourceText.slice(sourceCode.getFirstToken(blockBody[0]).range[1], returnValue.range[0]);
+                            const returnValueText = returnValue.type === "ObjectExpression" ? `(${sourceCode.getText(returnValue)})` : sourceCode.getText(returnValue);
+                            const textAfterValue = sourceText.slice(returnValue.range[1], blockBody[0].range[1] - 1);
+                            const textAfterReturnStatement = sourceText.slice(blockBody[0].range[1], arrowBody.range[1] - 1);
 
-                    if (asNeeded && blockBody[0].type === "ReturnStatement") {
-                        context.report({
-                            node,
-                            loc: arrowBody.loc.start,
-                            message: "Unexpected block statement surrounding arrow body."
-                        });
-                    }
+                            return fixer.replaceText(arrowBody, textBeforeReturn + textBetweenReturnAndValue + returnValueText + textAfterValue + textAfterReturnStatement);
+                        }
+                    });
                 }
             } else {
                 if (always || (asNeeded && requireReturnForObjectLiteral && arrowBody.type === "ObjectExpression")) {
                     context.report({
                         node,
                         loc: arrowBody.loc.start,
-                        message: "Expected block statement surrounding arrow body."
+                        message: "Expected block statement surrounding arrow body.",
+                        fix(fixer) {
+                            const lastTokenBeforeBody = sourceCode.getTokensBetween(sourceCode.getFirstToken(node), arrowBody)
+                                .reverse()
+                                .find(token => token.value !== "(");
+
+                            const firstBodyToken = sourceCode.getTokenAfter(lastTokenBeforeBody);
+
+                            return fixer.replaceTextRange(
+                                [firstBodyToken.range[0], node.range[1]],
+                                `{return ${sourceCode.getText().slice(firstBodyToken.range[0], node.range[1])}}`
```

Comment:
I thought of doing that, but that could cause comments to get removed:

```js
var foo = () => ( /* comment */ {ok: true});

// gets fixed to:

var foo = () => {return ( /* comment */ {ok: true});};
```

I think users can use `no-extra-parens` if they want to avoid the unnecessary parens.